Dockerfile

```
@@ -1,44 +1,94 @@
 
-#Print all logs without buffering it.
-ENV PYTHONUNBUFFERED 1
+# TODO: use --frozen-lockfile
+# RUN yarn install --frozen-lockfile
+RUN yarn install
 
-ENV DOCKER true
+COPY --link ./vue3 ./
+COPY --link ./vue ../vue
 
-#This port will be used by gunicorn.
-EXPOSE 8080
+FROM node-deps AS node-builder
+
+RUN <<EOF
+  sed -i -e 's/..\/cookbook\/static\/vue3/.\/build/g' ./vite.config.ts
+  yarn build
+EOF
 
-#Create app dir and install requirements.
-RUN mkdir /opt/recipes
+FROM python:3.13-alpine3.21 AS base
 WORKDIR /opt/recipes
 
-COPY requirements.txt ./
-
-RUN \
-    if [ `apk --print-arch` = "armv7" ]; then \
-    printf "[global]\nextra-index-url=https://www.piwheels.org/simple\n" > /etc/pip.conf ; \
-    fi
-
-# remove Development dependencies from requirements.txt
-RUN sed -i '/# Development/,$d' requirements.txt
-RUN apk add --no-cache --virtual .build-deps gcc musl-dev postgresql-dev zlib-dev jpeg-dev libwebp-dev openssl-dev libffi-dev cargo openldap-dev python3-dev xmlsec-dev xmlsec build-base g++ curl && \
-    echo -n "INPUT ( libldap.so )" > /usr/lib/libldap_r.so && \
-    python -m venv venv && \
-    /opt/recipes/venv/bin/python -m pip install --upgrade pip && \
-    venv/bin/pip debug -v && \
-    venv/bin/pip install wheel==0.45.1 && \
-    venv/bin/pip install setuptools_rust==1.10.2 && \
-    if [ `apk --print-arch` = "aarch64" ]; then \
-    curl https://sh.rustup.rs -sSf | sh -s -- -y; \
-    fi &&\
-    venv/bin/pip install -r requirements.txt --no-cache-dir &&\
-    apk --purge del .build-deps
-
-#Copy project and execute it.
-COPY . ./
+# Print all logs without buffering: https://stackoverflow.com/a/59812588
+ENV PYTHONUNBUFFERED=1
+
+# Install all dependencies.
+RUN apk add --no-cache \
+  gettext \
+  git \
+  libjpeg \
+  libwebp \
+  libxml2-dev \
+  libxslt-dev \
+  openldap \
+  postgresql-client \
+  postgresql-libs \
+  zlib
+
+FROM base AS deps
+
+ENV DOCKER=true
+
+COPY --link requirements.txt ./
+
+RUN <<EOF
+  if [ $(apk --print-arch) = "armv7" ]; then
+    printf "[global]\nextra-index-url=https://www.piwheels.org/simple\n" >/etc/pip.conf
+  fi
+
+  # remove Development dependencies from requirements.txt
+  sed -i '/# Development/,$d' requirements.txt
+
+  apk add --no-cache --virtual .build-deps \
+    build-base \
+    cargo \
+    curl \
+    g++ \
+    gcc \
+    jpeg-dev \
+    libffi-dev \
+    libwebp-dev \
+    musl-dev \
+    openldap-dev \
+    openssl-dev \
+    postgresql-dev \
+    python3-dev \
+    xmlsec \
+    xmlsec-dev \
+    zlib-dev
+
+  echo -n "INPUT ( libldap.so )" >/usr/lib/libldap_r.so
+
+  python -m venv venv
+  /opt/recipes/venv/bin/python -m pip install --upgrade pip
+  venv/bin/pip debug -v
+  venv/bin/pip install wheel==0.45.1
+  venv/bin/pip install setuptools_rust==1.10.2
+
+  if [ $(apk --print-arch) = "aarch64" ]; then
+    curl https://sh.rustup.rs -sSf | sh -s -- -y
+  fi
+
+  venv/bin/pip install -r requirements.txt --no-cache-dir
```

```suggestion
  venv/bin/pip install -r requirements.txt
```
related to the caching in the RUN command (proposed change)